docs/src/tutorial.jl

```diff
@@ -0,0 +1,309 @@
+
+animate_attractors_continuation(
+    ds, attractors_info, fractions_curves, prange, pidx;
+);
+
+# ```@raw html
+# <video width="auto" controls autoplay loop>
+# <source src="../attracont.mp4" type="video/mp4">
+# </video>
+# ```
+
+# Hah, how cool is that! The attractors pop in and out of existence like out of nowhere!
+# It would be incredibly difficult to find these attractors in traditional continuation software
+# where a rough estimate of the period is required! (It would also be too hard due to the presence
+# of chaos for most of the parameter values, but that's another issue!)
+
+# Now typically a continuation is visualized in a 2D plot where the x axis is the
+# parameter axis. We can do this with the convenience function:
+
+fig = plot_basins_attractors_curves(
+	fractions_curves, attractors_info, A -> minimum(A[:, 1]), prange,
+)
+
+# In the top panel are the basin fractions, by default plotted as stacked bars.
+# Bottom panel is a visualization of the tracked attractors.
+# The argument `A -> minimum(A[:, 1])` is simply a function tha maps
+# an attractor into a real number for plotting.
+# We can provide more functions to visualize more aspects of the attractors:
+
+using Statistics: std
+
+a2rs = [
+    A -> minimum(A[:, 1]),
+    A -> log(length(A)), # proxy for "complexity"
+]
+
+fig = plot_basins_attractors_curves(
+	fractions_curves, attractors_info, a2rs, prange; add_legend = false
+)
+
+ax1, ax2 = content.((fig[2,1], fig[3,1]))
+
+ax1.ylabel = "min(A₁)"
+ax2.ylabel = "log(len(A))"
+
+fig
+
+# ## Enhancing the continuation
+
+# The biggest strength of Attractors.jl is that it is not an isolated software.
+# It is part of **DynamicalSystems.jl**. Here, we will use the full power of
+# **DynamicalSystems.jl** and enrich the above continuation with various other
+# measures of nonlocal stability, in particular Lyapunov exponents and
+# the minimal fatal shock.
+
+# First, let's estimate the maximum Lyapunov exponent (MLE) for all attractors,
+# using the `lyapunovspectrum` function that comes from the ChaosTools.jl submodule.
+
+using ChaosTools: lyapunov
+
+lis = map(enumerate(prange)) do (i, p) # loop over parameters
+    set_parameter!(ds, pidx, p) # important! We use the dynamical system!
+    attractors = attractors_info[i]
+    Dict(k => lyapunov(ds, 2000.0; u0 = A[1]) for (k, A) in attractors)
+end
+
+# The above `map` loop may be intimidating if you are a beginner, but it is
+# really just a shorter way to write a `for` loop for our example.
+# We iterate over all parameters, and for each we first update the dynamical
+# system with the correct parameter, and then extract the MLE
+# for each attractor. `map` just means that we don't have to pre-allocate a
+# new vector before the loop; it creates it for us.
+
+# We can visualize the LE with the other convenience function [`plot_continuation_curves!`](@ref),
+
+ax3 = Axis(fig[4, 1]; ylabel = "MLE")
+plot_continuation_curves!(ax3, lis, prange; add_legend = false)
+
+fig
```

Comment:
This graphic is really cool. Is there a simple way to stretch the height? It looks cluttered because the plots are too close.

Reply:
I stretch once at the end it's fine for now!